public/app/plugins/datasource/prometheus/query_hints.test.ts

```diff
@@ -46,21 +18,21 @@ describe('getQueryHints()', () => {
         ],
       },
     ];
-    const hints = getQueryHints('metric', series);
+    const hints = getQueryHints('metric_total', series);
```

Comment:
I'm not sure these test cases cover the new code we've introduced. I would expect us to pass in a stubbed data source with the different scenarios (e.g. metadata available, unmatched, etc.).

That being said, I'm not familiar with the testing structure of the data sources, and it might be a case that we get enough confidence from typescript + manual testing. I'll leave that decision in your hands.